src/pages/patientView/trialMatch/TrialMatchTableUtils.ts

```diff
@@ -0,0 +1,129 @@
+        let positiveCancerTypes: string[] = [];
+        let negativeCancerTypes: string[] = [];
+        _.map(cancerTypes, (item) => {
+            // If a cancer type contains a "!", it means this trial cannot be used for the cancer type, which is a "NOT" match.
+            if (item.includes('!')) {
+                negativeCancerTypes.push(item.replace('!', ''));
+            } else {
+                positiveCancerTypes.push(item);
+            }
+        });
+        const clinicalGroupMatch: IClinicalGroupMatch = {
+            trialAgeNumerical: age,
+            trialOncotreePrimaryDiagnosis: {
+                positive: positiveCancerTypes,
+                negative: negativeCancerTypes
+            },
+            matches: [],
+            notMatches: []
+        };
+        groupTrialMatchesByGenomicAlteration(matchesGroupedByAge[age], clinicalGroupMatch);
+        matchedTrial.priority += setTrialMatchPriority(clinicalGroupMatch);
+        armMatch.matches.push(clinicalGroupMatch);
+    });
+}
+
+export function groupTrialMatchesByGenomicAlteration(ageGroup: ITrialMatch[], clinicalGroupMatch: IClinicalGroupMatch) {
+    const matchesGroupedByGenomicAlteration = _.groupBy(ageGroup, (trial: ITrialMatch) => trial.genomicAlteration);
+    _.forEach(matchesGroupedByGenomicAlteration, (genomicAlterationGroup, genomicAlteration) => {
+        const genomicGroupMatch = groupTrialMatchesByPatientGenomic(genomicAlterationGroup, genomicAlteration);
+        // If a genomic alteration(i.e. BRAF V600E) contains a "!",
+        // it means this trial cannot be used for the genomic alteration, which is a "NOT" match.
+        if(genomicAlteration.includes('!')) {
+            clinicalGroupMatch.notMatches.push(genomicGroupMatch);
+        } else {
+            clinicalGroupMatch.matches.push(genomicGroupMatch);
+        }
+    });
+}
+
+export function groupTrialMatchesByPatientGenomic(genomicAlterationGroup: ITrialMatch[], genomicAlteration: string): IGenomicGroupMatch {
+    const matchesGroupedByPatientGenomic = _.groupBy(genomicAlterationGroup, (trial: ITrialMatch) => trial.patientGenomic);
+    const genomicGroupMatch: IGenomicGroupMatch = {
+        genomicAlteration: genomicAlteration,
+        matches: []
+    };
+    _.forEach(matchesGroupedByPatientGenomic , (patientGenomicGroup: ITrialMatch[]) => {
+        const genomicMatch: IGenomicMatch = {
+            trueHugoSymbol: patientGenomicGroup[0].trueHugoSymbol,
+            trueProteinChange: patientGenomicGroup[0].trueProteinChange,
+            sampleIds: _.uniq(_.map(patientGenomicGroup, (trial: ITrialMatch) => trial.sampleId))
+        };
+        genomicGroupMatch.matches.push(genomicMatch);
+    });
+    return genomicGroupMatch;
+}
+
+export function setTrialMatchPriority(clinicalGroupMatch: IClinicalGroupMatch): number {
+    if (clinicalGroupMatch.notMatches.length > 0) {
```

Comment:
would be nice to have a comment explaining this business logic here

Reply:
Add comments.